src/ResourceManager/Version2016_09_01/Utilities/Models/ResourceIdentifier.cs

```diff
@@ -32,42 +33,59 @@ public class ResourceIdentifier
 
         public ResourceIdentifier() { }
 
+        private class UrlTagName
+        {
+            public const string subscriptions = "subscriptions",
```

Comment:
I don't think we need to add this assumption. If we need to add it, please compare string in case insensitive